src/internal/cache/redisclient.go

```diff
@@ -0,0 +1,145 @@
+package cache
+
+import (
+	"fmt"
+	"github.com/FZambia/sentinel"
+	"github.com/goharbor/harbor/src/common/utils/log"
+	"github.com/gomodule/redigo/redis"
+	"net/url"
+	"regexp"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+)
+
+var knownPool sync.Map
+var m sync.Mutex
+var redisPathDBRegexp = regexp.MustCompile(`/(\d*)\z`)
+
+// RedisPoolParam ...
+type RedisPoolParam struct {
+	PoolMaxIdle     int
+	PoolMaxActive   int
+	PoolIdleTimeout time.Duration
+
+	DialConnectionTimeout time.Duration
+	DialReadTimeout       time.Duration
+	DialWriteTimeout      time.Duration
+}
+
+// GetRedisPool get a named redis pool
+// supported schema
+// redis://user:pass@redis_host:port/db
+// redis+sentinel://user:pass@redis_sentinel1:port1,redis_sentinel2:port2/monitor_name/db
+func GetRedisPool(name string, rawurl string, param *RedisPoolParam) (*redis.Pool, error) {
```

Comment:
This func is a little big, you can split it into several small ones. For normal redis pool and for sentinel pool etc.

Reply:
ok, will do it